lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_l10n_helper.rb

```diff
@@ -0,0 +1,178 @@
+      
+          config = {
+            'input_dir' => input_dir,
+            'output_dir' => output_dir,
+            'strings' => langs.map do |lang|
+              {
+                  'inputs' => ["#{lang}.lproj/Localizable.strings"],
+                  'options' => { 'separator' => "____" }, # Choose an unlikely one to avoid creating needlessly complex Stencil Context due to '.' in sentences sued as keys
+                  'outputs' => [{
+                      'templatePath' => template_path,
+                      'output' => output_filename(lang)
+                  }]
+              }
+            end
+          }
+      
+          # Write SwiftGen config file
+          config_file = File.join(output_dir, CONFIG_FILE_NAME)
+          File.write(config_file, config.to_yaml)
+      
+          return [config_file, langs]
+        end
+
+        # Because we use English copy verbatim as key names, some keys are the same just except for the upper/lowercase.
+        # We need to sort the output again because SwiftGen only sort case-insensitively so that means keys that are
+        # the same except case might be in swapped order for different outputs
+        #
+        # @param [String] dir The temporary directory in which the file to sort lines for is located
+        # @param [String] lang The code for the locale we need to sort the output lines for
+        #
+        def sort_file_lines!(dir, lang)
+          file = File.join(dir, output_filename(lang))
+          sorted_lines = File.readlines(file).sort
+          File.write(file, sorted_lines.join)
+          return file
+        end
+
+        # Prepares the template and config files, then run SwiftGen, run `diff` on each generated output against the baseline, and returns a Hash of the violations found.
+        #
+        # @param [String] input_dir The directory where the `.lproj` folders to scan are located
+        # @param [String] base_lang The base language used as source of truth that all other languages will be compared against
+        # @return [Hash<String, String>] A hash of violations, keyed by language code, whose values are the diff output.
+        #
+        # @note The returned Hash contains keys only for locales with violations. Locales parsed but without any violations found will not appear in the resulting hash.
+        #
+        def find_diffs(input_dir:, base_lang:)
+          Dir.mktmpdir('a8c-lint-translations-') do |tmpdir|
+            # Run SwiftGen 
+            (config_file, langs) = generate_swiftgen_config!(input_dir, tmpdir)
+            Action.sh(swiftgen_bin, 'config', 'run', '--config', config_file)
+            
+            # Run diffs
+            base_file = sort_file_lines!(tmpdir, base_lang)
+            langs.delete(base_lang)
+            return Hash[langs.map do |lang|
+              file = sort_file_lines!(tmpdir, lang)
+              diff = `diff -U0 "#{base_file}" "#{file}"`
+              diff.gsub!(/^(---|\+\+\+).*\n/, '')
```

Comment:
Would it be worth adding a comment on this substitution of the diff file names? Also regarding the fact that the `@@ from-file-line-numbers to-file-line-numbers @@` is retained so that users can find the issues faster?